### ISSUE TITLE
Add "expandable" property defining if row can be expanded

### ISSUE THREAD

**Issue by camdarley** [icflorescu/mantine-datatable#579]
**Is your feature request related to a problem? Please describe.**
I would like to use the expansion feature with nested tables, to expand rows which have a children property.
For rows which don't have children, I would like to disable the expansion feature.

**Describe the solution you'd like**
Best would be to add a `expandable` prop, which is a function taking record and index as arguments, and returning a boolean defining wether or not the row is expandable.
```
<DataTable
      withTableBorder
      withColumnBorders
      columns={[{ accessor: 'name' }, { accessor: 'city' }, { accessor: 'state' }]}
      records={records}
      rowExpansion={{
        allowMultiple: true,
        expandable: ({ record }) => record.children && record.children.length > 0, // 👈 enable expansion only on rows which have children
        content: ({ record }) => (
        ...
```

**Comment by camdarley** [comment]
#580

**Comment by icflorescu** [comment]
That's a good idea, thanks.
I'm up to my ears in a project at the moment, but I'll have a look at how this could be implemented when I'll find a moment.

**Comment by icflorescu** [comment]
Wow, that was fast! :-)

**Comment by camdarley** [comment]
Actually I filled the issue while I was already working on the PR :-)

**Comment by icflorescu** [comment]
Thanks, I'll merge and publish it later today!

**Comment by camdarley** [comment]
Great, thanks!
I just pushed a new commit with default behavior (when expandable prop is missing) set as "expandable", otherwise it would break existing code!

**Comment by icflorescu** [comment]
Released in `v7.8.2` 🎉
